cfme/infrastructure/virtual_machines.py

```diff
@@ -993,18 +993,25 @@ def _method_setup(vm_names, provider_crud=None):
         sel.check(Quadicon(vm_name, 'vm').checkbox())
 
 
-def find_quadicon(vm_name):
+def find_quadicon(vm_name, do_not_navigate=False):
     """Find and return a quadicon belonging to a specific vm
 
     Args:
         vm: vm name as displayed at the quadicon
-    Returns: entity of appropriate class
+    Returns: :py:class:`cfme.web_ui.Quadicon` instance
     """
-    # todo: VMs have such method, so, this function is good candidate for removal
-    view = navigate_to(Vm, 'VMsOnly')
-    try:
-        return view.entites.get_entity(vm_name, surf_pages=True)
-    except ItemNotFound:
+    if not do_not_navigate:
```

Comment:
REQUIRED: please get_rid of web_ui in your new code. 
btw, you are going to replace widgetastic code, I've recently added with old web_ui. Please double check diffs during rebase.